src/server/services/wfs/qgswfsgetfeature.cpp

```diff
@@ -461,7 +461,20 @@ namespace QgsWfs
       {
 
         // For WFS 1.1 we honor requested CRS and axis order
-        const QString srsName {request.serverParameters().value( QStringLiteral( "SRSNAME" ) )};
+        // Axis is not inverted if srsName starts with EPSG
+        // It needs to be an EPSG urn, e.g. urn:ogc:def:crs:EPSG::4326
+        // This follows geoserver convention
+        // See: https://docs.geoserver.org/stable/en/user/services/wfs/axis_order.html
+        // if the crs is defined in the parameters, use it
+        // otherwise:
+        //  - geojson uses 'EPSG:4326' by default
+        //  - other formats use the layer's crs
```

Comment:
Sorry but this doesn't sound right to me: if the layer's CRS is not one of the advertized CRSs it should not be used.

I think I made it clear in my previous comment.

Reply:
This PR is not ready yet. I should have mentioned it. My bad. While working on it, I noticed a regression in some geojson output tests. I pushed a temporary version to check that the tests passed.

Reply:
Do you know you can convert a PR to Draft ?

Reply:
Indeed. Done.